sources/shared-defaults/kubernetes-aws-credential-provider.toml

```diff
@@ -4,9 +4,15 @@ cache-duration = "12h"
 image-patterns = [
     "*.dkr.ecr.*.amazonaws.com",
     "*.dkr.ecr.*.amazonaws.com.cn",
+    "*.dkr.ecr.*.on.aws",
+    "*.dkr.ecr.*.on.amazonwebservices.com.cn",
```

Comment:
Do we need a migration to add these new patterns on upgrade?

If we do, we should pull in https://github.com/bottlerocket-os/bottlerocket/pull/4419 also. CC: @ytsssun

Reply:
Bah, you're right, we do. I'll add one.

I think it makes sense to tackle both with a single throw if we can.